explorer/app/components/Table/table.tsx

```diff
@@ -13,31 +13,50 @@ interface TableProps<T extends object> {
   items: T[];
   pageSize: number;
   columns: Column<T>[];
+  total?: number;
+  onNextPageClicked?: () => void;
+  onPreviousPageClicked?: () => void;
+  canPreviousPage?: boolean;
+  canNextPage?: boolean;
+  currentPage?: number;
 }
 
+/**
+ * This table can be Controlled or Uncrontrolled based on the set of props passed to it.
+ * Controlled table will receive the navigation functions and it will be used for dynamic loads.
+ * Uncrontrolled table will take advantage of react's table state and will be used for static loads.
+ */
 export const Table = <T extends object>({
   items,
   columns,
   pageSize,
+  total,
+  onNextPageClicked,
+  onPreviousPageClicked,
+  canPreviousPage,
+  canNextPage,
+  currentPage,
 }: TableProps<T>): JSX.Element => {
   const {
-    nextPage,
-    previousPage,
     getTableProps,
     headers,
     getTableBodyProps,
     page,
     prepareRow,
-    canPreviousPage,
-    canNextPage,
+    canNextPage: tableCanNextPage,
+    canPreviousPage: tableCanPreviousPage,
+    nextPage: tableNextPage,
+    previousPage: tablePreviousPage,
     pageOptions,
     state: { pageIndex },
   } = useTable<T>(
     {
       columns,
       data: items,
+      manualPagination: !!currentPage,
+      pageCount: total,
       initialState: {
-        pageSize: pageSize,
+        pageSize: pageSize ?? 25,
```

Comment:
Can you create a constant (local to this file) for this default page size?

Reply:
Actually, taking a second look, is the nullish coalescing necessary here? It looks like `pageSize` is a required prop.

Reply:
oh right! This isn't necessary